subprojects/docs/src/samples/maven/pomGeneration/groovy/build.gradle

```diff
@@ -64,6 +65,7 @@ uploadArchives {
 // Customize the contents of the pom
 
 // tag::when-configured[]
+// FIXME: Convert to lazy task
 def installer = install.repositories.mavenInstaller
 def deployer = uploadArchives.repositories.mavenDeployer
```

Comment:
How should this be migrated?

```
def mavenInformation = [
    'installer': 'mavenInstaller',
    'uploadArchives': 'mavenDeployer'
]
def installer = tasks.named("install")
def deployer = tasks.named("uploadArchives")
[installer, deployer]*.configure {
    repositories."${mavenInformation[name]}".pom.whenConfigured { ... }
    repositories."${mavenInformation[name]}".pom.version = '1.0MVN'
}
installer.configure {
    repositories.mavenInstaller.pom.project { ... }
}
deployer.configure {
    repositories.mavenDeployer.pom.groupId = 'deployGroup'
}
```

Reply:
I'd extract a function instead of relying on string references:
```groovy
def configurePom(MavenResolver resolver) {
    resolver.pom.whenConfigured { pom ->
        pom.dependencies.find {dep -> dep.groupId == 'group3' && dep.artifactId == 'runtime' }.optional = true
    }
}
tasks.named("install", Upload) {
    configurePom(repositories.mavenInstaller)
}
tasks.named("uploadArchives", Upload) {
    configurePom(repositories.mavenDeployer)
}
```